File: user-service/src/main/java/rs/raf/user_service/client/StockClient.java
package rs.raf.user_service.client;

import org.springframework.cloud.openfeign.FeignClient;
import org.springframework.web.bind.annotation.GetMapping;
import rs.raf.user_service.domain.dto.OrderDto;

import java.util.List;

@FeignClient(name = "stock-service", url = "${spring.cloud.openfeign.client.config.stock-service.url}")
public interface StockClient {
    @GetMapping("api/orders/all")
    List<OrderDto> getAll();
}

File: user-service-kotlin/src/main/kotlin/pack/userservicekotlin/external/StockClient.kt
package pack.userservicekotlin.external

import org.springframework.cloud.openfeign.FeignClient
import org.springframework.web.bind.annotation.GetMapping
import pack.userservicekotlin.domain.dto.external.OrderDto

@FeignClient(
    name = "stock-service",
    url = "\${spring.cloud.openfeign.client.config.stock-service.url}",
)
interface StockClient {
    @GetMapping("api/orders/all")
    fun getAll(): List<OrderDto>
}
